Edit `apt-get` installation instructions

ptgott opened the pull request:
Closes #24564

Show using `/etc/apt/trusted.gpg.d`, a directory that `apt-get` searches automatically, for storing public keys. Edit the three locations where we showed using `/usr/share/keyrings`, including two partials.

marcoandredinis commented:
Using the `[signed-by=<pubkey>]` in the `/etc/apt/sources.list.d/teleport.list` file, we ensure only that key is considered when validating the teleport deb package.

By removing this option, we are saying that any globally installed pub key in `/etc/apt/trusted.gpg.d/` is considered when validating the teleport deb package.
It changes things slightly in terms of security invariants.

I'm raising this topic because I'm also interested in applying this change to Server Auto Discover scripts and JoinScript.
If this PR is merged, I'll change those places as well.

ptgott commented:
> Using the `[signed-by=<pubkey>]` in the `/etc/apt/sources.list.d/teleport.list` file, we ensure only that key is considered when validating the teleport deb package.
> 
> By removing this option, we are saying that any globally installed pub key in `/etc/apt/trusted.gpg.d/` is considered when validating the teleport deb package. It changes things slightly in terms of security invariants.
> 
> I'm raising this topic because I'm also interested in applying this change to Server Auto Discover scripts and JoinScript. If this PR is merged, I'll change those places as well.

Thanks for the context! I've responded to your feedback in [b9c926b](https://github.com/gravitational/teleport/pull/50033/commits/b9c926b6cad1df8c331e87bb9620e3f0ebd74967).

zmb3 reviewed (approved):
LGTM if Marco is good with it. Not my area of expertise.

ptgott commented:
@marcoandredinis Just double-checking if this is okay. Thanks!

marcoandredinis commented:
After reading this comment https://github.com/gravitational/teleport/pull/50273#pullrequestreview-2526280906 cc @gzdunek 
I think we should download the public key into `/etc/apt/keyrings/` (we might need to add `mkdir -p /etc/apt/keyrings/` before writing to it)

marcoandredinis commented:
@ptgott It seems that you marked the PR as draft. Do you want me to follow up on this one? Or is there anything blocking the last suggestion?

ptgott committed:
Edit apt-get installation instructions

Closes #24564

Show using `/etc/apt/trusted.gpg.d`, a directory that `apt-get` searches
automatically, for storing public keys. Edit the three locations where
we showed using `/usr/share/keyrings`, including two partials.

Trust only the Teleport-issued public key when validating Teleport DEB
packages.

ptgott committed:
Use /etc/apt/keyrings for apt public keys

Respond to marcoandredinis feedback.

ptgott commented:
> After reading this comment [#50273 (review)](https://github.com/gravitational/teleport/pull/50273#pullrequestreview-2526280906) cc @gzdunek I think we should download the public key into `/etc/apt/keyrings/` (we might need to add `mkdir -p /etc/apt/keyrings/` before writing to it)

@marcoandredinis I wanted to make sure this wouldn't get merged while I prioritized other things, but now I've made the change and this is ready for another look. Thanks!
